lib/grains_cases.rb

```diff
@@ -0,0 +1,71 @@
+class GrainsCase < OpenStruct
+  def test_name
+    'test_%s' % description.downcase.tr_s(' ', '_')
+  end
+
+  def workload
+    assertion
+  end
+
+  def skipped
+    index.zero? ? '# skip' : 'skip'
+  end
+end
+
+class GrainsCase::SquareAssertion
+  def initialize(expected:, input:)
+    @expected = expected
+    @input = input
+  end
+
+  def to_s
+    return error_assertion if expected < 0
+
+    "assert_equal #{expected}, Grains.square(#{input})"
+  end
+
+  private
+
+  attr_reader :expected, :input
+
+  def error_assertion
+      "assert_raises(ArgumentError) { Grains.square(#{input}) }"
+  end
+end
+
+class GrainsCase::TotalAssertion
+  def initialize(expected:)
+    @expected = expected
+  end
+
+  def to_s
+    "assert_equal #{expected}, Grains.total"
+  end
+
+  private
+
+  attr_reader :expected
+end
+
+GrainsCases = proc do |data|
+  data = JSON.parse(data)
+
+  cases = data['square']['cases'].map.with_index do |row, i|
+    assertion = GrainsCase::SquareAssertion.new(
+      expected: row['expected'],
+      input: row['input'],
+    )
+
+    GrainsCase.new(row.merge('assertion' => assertion, 'index' => i))
+  end
+
+  assertion = GrainsCase::TotalAssertion.new(
```

Comment:
This is because of the format of the canonical data:
```json
{
  "square": {
    "cases": [
      { "description": "...", "input": 1, "expected": 1 },
      ...
    ]
  },
  "total": {
    "description": "...",
    "expected": "..."
  }
}
```